pkg/compute/compute_integration_test.go

```diff
@@ -399,92 +571,166 @@ func TestBuildRust(t *testing.T) {
 		},
 		{
-			name:           "unknown language",
-			args:           []string{"compute", "build"},
-			fastlyManifest: "name = \"test\"\nlanguage = \"javascript\"\n",
-			client:         versionClient{fastlyVersions: []string{"0.0.0"}},
-			wantError:      "unsupported language javascript",
+			name: "unknown language",
+			args: []string{"compute", "build"},
+			fastlyManifest: `
+			name = "test"
+			language = "foobar"`,
+			client:    versionClient{fastlyVersions: []string{"0.0.0"}},
+			wantError: "unsupported language foobar",
 		},
 		{
-			name:           "error reading cargo metadata",
-			args:           []string{"compute", "build"},
-			fastlyManifest: "name = \"test\"\nlanguage = \"rust\"\n",
-			cargoManifest:  "[package]\nname = \"test\"",
-			client:         versionClient{fastlyVersions: []string{"0.4.0"}},
-			wantError:      "reading cargo metadata",
+			name: "error reading cargo metadata",
+			args: []string{"compute", "build"},
+			applicationConfig: config.File{
+				Language: config.ConfigLanguage{
+					Rust: config.ConfigRust{
+						// TODO: pull actual version from .github/workflows/pr_test.yml
+						// when doing local run of integration tests.
+						ToolchainVersion: "1.49.0",
+						WasmWasiTarget:   "wasm32-wasi",
+					},
+				},
+			},
+			fastlyManifest: `
+			name = "test"
+			language = "rust"`,
+			cargoManifest: `
+			[package]
+			name = "test"`,
+			client:    versionClient{fastlyVersions: []string{"0.4.0"}},
+			wantError: "reading cargo metadata",
 		},
 		{
-			name:           "fastly-sys crate not found",
-			args:           []string{"compute", "build"},
-			fastlyManifest: "name = \"test\"\nlanguage = \"rust\"\n",
-			cargoManifest:  "[package]\nname = \"test\"\nversion = \"0.1.0\"\n\n[dependencies]\nfastly = \"=0.3.2\"",
-			cargoLock:      "[[package]]\nname = \"test\"\nversion = \"0.1.0\"\n\n[[package]]\nname = \"fastly\"\nversion = \"0.3.2\"",
+			name: "fastly-sys crate not found",
+			args: []string{"compute", "build"},
+			applicationConfig: config.File{
+				Language: config.ConfigLanguage{
+					Rust: config.ConfigRust{
+						ToolchainVersion:    "1.49.0",
+						WasmWasiTarget:      "wasm32-wasi",
+						FastlySysConstraint: "0.0.0",
+					},
+				},
+			},
+			fastlyManifest: `
+			name = "test"
+			language = "rust"`,
+			cargoManifest: `
+			[package]
+			name = "test"
+			version = "0.1.0"
+			[dependencies]
+			fastly = "=0.3.2"`,
+			cargoLock: `
+			[[package]]
+			name = "test"
+			version = "0.1.0"
+			[[package]]
+			name = "fastly"
+			version = "0.3.2"`,
 			client: versionClient{
-				fastlyVersions:    []string{"0.4.0"},
-				fastlySysVersions: []string{"0.0.0"}, // included to stop REST API failing to not find crate
+				fastlyVersions: []string{"0.4.0"},
 			},
-			wantError:            "fastly-sys crate not found",
+			wantError:            "fastly-sys crate not found", // because fastly <0.4.0 has no fastly-sys
```

Comment:
`fastly 0.3.3` is where `fastly-sys` was introduced, but the earlier `0.3.x` do not have it.